### PR TITLE
Split ilayer/131

### PR DESCRIPTION
**Pull request by molleer**

### CODE REVIEW

File: src/main/java/chalmers/pimp/model/Point.java [hunk 1]
```diff
@@ -92,6 +92,18 @@ public Point addY(int yOffset) {
     return new Point(x, y + yOffset);
   }
 
+  /**
+   * Returns the distance between this and point p
+   *
+   * @param p the point to be compared with
+   * @return the distance this point and point p
```
Comment (theowiik) [inline src/main/java/chalmers/pimp/model/Point.java]: null point?

File: src/main/java/chalmers/pimp/model/canvas/layer/Doodle.java [hunk 2]
```diff
@@ -241,4 +235,18 @@ private int getExtreme(List<? extends Integer> list,
     }
     return extreme;
   }
+
+  @Override
+  public void addPoint(Point p) {
```
Comment (theowiik) [inline src/main/java/chalmers/pimp/model/canvas/layer/Doodle.java]: null point?

File: src/main/java/chalmers/pimp/model/canvas/layer/IDoodleLayer.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package chalmers.pimp.model.canvas.layer;
+
+import chalmers.pimp.model.Point;
+import chalmers.pimp.model.viewport.IReadOnlyViewport;
+
+public interface IDoodleLayer extends ILayer {
+
+  /**
+   * Adds a point which is used wile rendering
```
Comment (theowiik) [inline src/main/java/chalmers/pimp/model/canvas/layer/IDoodleLayer.java]: typo, change to "while"

File: src/main/java/chalmers/pimp/model/canvas/layer/Doodle.java [hunk 4]
```diff
@@ -241,4 +235,18 @@ private int getExtreme(List<? extends Integer> list,
...
+    points.add(p);
+  }
+
+  @Override
+  public void removePoint(Point p, double threshold) {
```
Comment (theowiik) [inline src/main/java/chalmers/pimp/model/canvas/layer/Doodle.java]: null point

File: src/main/java/chalmers/pimp/model/canvas/layer/IRasterLayer.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package chalmers.pimp.model.canvas.layer;
+
+import chalmers.pimp.model.pixeldata.IPixel;
+
+public interface IRasterLayer extends ILayer {
```
Comment (theowiik) [inline src/main/java/chalmers/pimp/model/canvas/layer/IRasterLayer.java]: missing javadoc

File: src/main/java/chalmers/pimp/model/canvas/layer/Raster.java [hunk 6]
```diff
@@ -12,7 +12,7 @@
  * The {@code Raster} class is an implementation of the {@code ILayer} interface that represents a
```
Comment (theowiik) [inline src/main/java/chalmers/pimp/model/canvas/layer/Raster.java]: change "implementation of the ILayer" interface to IRasterLayer

File: src/main/java/chalmers/pimp/model/tools/DoodleTool.java [hunk 7]
```diff
@@ -42,7 +44,8 @@ public void dragged(MouseStatus mouseStatus) {
     int x = model.getViewport().getTranslatedX(mouseStatus.getX());
     int y = model.getViewport().getTranslatedY(mouseStatus.getY());
 
-    doodle.setPixel(PixelFactory.createPixel(x, y));
+    doodle.addPoint(new Point(mouseStatus.getX() + model.getViewport().getX(),
+        mouseStatus.getY() + model.getViewport().getY()));
```
Comment (theowiik) [inline src/main/java/chalmers/pimp/model/tools/DoodleTool.java]: move out to local variable

File: src/main/java/chalmers/pimp/model/canvas/layer/Doodle.java (hunk 4 above)
Comment (molleer) [reply src/main/java/chalmers/pimp/model/canvas/layer/Doodle.java]: requireNotNull is called in the .distance method

File: src/main/java/chalmers/pimp/model/Point.java [hunk 8]
```diff
@@ -92,6 +92,19 @@ public Point addY(int yOffset) {
...
+   */
```
Comment (theowiik) [inline src/main/java/chalmers/pimp/model/Point.java]: specify in javadoc that it throws nullpointerexception

File: src/main/java/chalmers/pimp/model/canvas/layer/IDoodleLayer.java [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+   * @throws NullPointerException if the point is {@code null}
+   */
+  void addPoint(Point p);
+
+  /**
+   * Removes all points within threshold distance of the specified point
+   *
+   * @param p         the specified point
+   * @param threshold the threshold distance in pixels
+   */
```
Comment (theowiik) [inline src/main/java/chalmers/pimp/model/canvas/layer/IDoodleLayer.java]: javadoc for if it throws null pointer

**Pull request merged**